### ISSUE TITLE
Should Data objects convert stat/syserror to NumPy array too?

### ISSUE THREAD

**Issue by DougBurke** [sherpa/sherpa#703]
[this is not an issue that needs addressing for the 4.12.0 release]

With the master branch (what will be in the Sherpa 4.12.0 release), we now have the following behavior from the data objects:

```
>>> from sherpa.data import Data1D
>>> d = Data1D('x', [1,2], [2,3], syserror=[4,5])
/home/dburke/sherpa/sherpa-411/sherpa/data.py:49: UserWarning: Converting array [2, 3] to numpy array
  warnings.warn("Converting array {} to numpy array".format(array))
>>> print(d)
name      = x
x         = Int64[2]
y         = Int64[2]
staterror = None
syserror  = [4, 5]
>>> d = Data1D('x', [1,2], [2.1,3.1], staterror=[4,5])
/home/dburke/sherpa/sherpa-411/sherpa/data.py:49: UserWarning: Converting array [2.1, 3.1] to numpy array
  warnings.warn("Converting array {} to numpy array".format(array))
>>> print(d)
name      = x
x         = Int64[2]
y         = Float64[2]
staterror = [4, 5]
syserror  = None
```

Note that the `x` and `y` attribtes are converted to NumPy arrays and displayed as `<data type>[number of elements]` form in the string output (there's also a warning about the dependent axis - `y` in this case - being converted to a NumPy array, but that's not what I'm talking about here). I believe this was part of the data-rework in #614 - since in the 4.11 release we see

```
>>> from sherpa.data import Data1D
>>> d = Data1D('x', [1,2], [2,3], syserror=[4,5])
>>> print(d)
name      = x
x         = [1, 2]
y         = [2, 3]
staterror = None
syserror  = [4, 5]
>>> d = Data1D('x', [1,2], [2.1,3.1], staterror=[4,5])
>>> print(d)
name      = x
x         = [1, 2]
y         = [2.1, 3.1]
staterror = [4, 5]
syserror  = None
```

My question is whether we should convert `staterror` and/or `syserror` values - if given - to NumPy arrays in the data constructor? I believe we should (there's been at least one case when I was reworing some stats code when it would have been a little easier if we could just assume everything was a NumPy array), and it would match up with the `x`/`y` changes. That is, we should see something like

```
>>> d = Data1D('x', [1, 2], [16.0, 25.0], [4.0, 5.0])
>>> print(d)
>>> print(d)
name      = x
x         = Int64[2]
y         = Float64[2]
staterror = Float64[2]
syserror  = None
```

Just to note that `sherpa.ui.load_arrays` does this conversion for us, so this is a question for users of the `Data*` classes - e.g. we have

```
>>> from sherpa.ui import load_arrays, get_data
>>> load_arrays(1, [1,2,3], [4,5,6], [2,3,4])
>>> print(get_data())
name      = 
x         = Int64[3]
y         = Int64[3]
staterror = Int64[3]
syserror  = None
```

**Comment by hamogu** [comment]
SDS has decided that any input to Dataclasses for x, y, x_lo, x_hi, sys_err, stat_err etc should be converted to numpy arrays in the constructor. People should use numpy arrays anyway and
we've implicitly claimed that the type of input does not matter for the
result, so we are fixing a bug here, not "breaking people's code who
relied on having a different answer when they put in the tuple instead
of a numpy array".

**Comment by DougBurke** [comment]
This was fixed, probably in #1477
